packages/next/src/build/webpack/plugins/nextjs-require-cache-hot-reloader.ts

```diff
@@ -96,35 +96,40 @@ export class NextJsRequireCacheHotReloader implements WebpackPluginInstance {
       clearModuleContext(targetPath)
     })
 
-    compiler.hooks.afterEmit.tap(PLUGIN_NAME, (compilation) => {
-      RUNTIME_NAMES.forEach((name) => {
-        const runtimeChunkPath = path.join(
-          compilation.outputOptions.path!,
-          `${name}.js`
-        )
-        deleteCache(runtimeChunkPath)
-      })
-      let hasAppPath = false
-
-      // we need to make sure to clear all server entries from cache
-      // since they can have a stale webpack-runtime cache
-      // which needs to always be in-sync
-      const entries = [...compilation.entries.keys()].filter((entry) => {
-        const isAppPath = entry.toString().startsWith('app/')
-        hasAppPath = hasAppPath || isAppPath
-        return entry.toString().startsWith('pages/') || isAppPath
-      })
-
-      if (hasAppPath) {
+    compiler.hooks.afterEmit.tapAsync(
+      PLUGIN_NAME,
+      async (compilation, callback) => {
+        try {
+          for (const name of RUNTIME_NAMES) {
+            const runtimeChunkPath = path.join(
+              compilation.outputOptions.path!,
+              `${name}.js`
+            )
+            await deleteCache(runtimeChunkPath)
+          }
+          let hasAppPath = false
+
+          // we need to make sure to clear all server entries from cache
+          // since they can have a stale webpack-runtime cache
+          // which needs to always be in-sync
+          const entries = [...compilation.entries.keys()].filter((entry) => {
```

Comment:
I think you can use `entry.getEntrypointChunk()` to get the entrypoint file without needing to manually create the name

Reply:
I'm not sure if we need this filter here anymore, didn't specifically change it in this PR though.